backend/municipal-services/ingestion-service/app/ingest/facility_template_service.py

```
@@ -99,4 +100,42 @@ def _format_boundary_data(self, boundary_data: List[Boundary]) -> List[Dict[str,
+
+            with pd.ExcelWriter(output_path, engine='openpyxl') as writer:
+                all_sheets = pd.read_excel(input_path, sheet_name=None)
+                for sheet, sheet_df in all_sheets.items():
+                    if sheet == sheet_name:
+                        df.to_excel(writer, sheet_name=sheet, index=False)
+                    else:
+                        sheet_df.to_excel(writer, sheet_name=sheet, index=False)
+
+        except ValueError as e:
+            if "No sheet named" in str(e):
+                raise HTTPException(status_code=400, detail=f"Sheet '{sheet_name}' not found in the uploaded file")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)